practice-two/src/App.jsx

```diff
@@ -25,6 +25,24 @@ const App = () => {
   // State for storing the favorite product IDs
   const [favoriteProductIDs, setFavoriteProductIDs] = useState([]);
 
+  // State for storing the selected product's ID for details view
+  const [selectedProductID, setSelectedProductID] = useState(null);
+
+  // Function to handle selecting a product card
+  const handleSelectProduct = (productID) => {
+    setSelectedProductID((prevProductID) => {
+      if (prevProductID === productID) {
+        return null; // Close the details if it's the same product
+      }
+      return productID; // Display the details of the selected product
```

Comment:
the same here. I don't see it's relevant to "Display the details of the selected product"

Reply:
Done